app/src/main/java/ch/epfl/sdp/entity/Inventory.java

```diff
@@ -20,62 +18,34 @@ public Inventory(Player player) {
         this.items = new HashMap<>();
     }
 
-    public void useItem(Item i) {
-        int numberOfInstances = items.get(i.getName());
-
-        if (numberOfInstances > 0) {
-            switch (i.getName()) {
+    public void useItem(String itemName) {
```

Comment:
Cool refactoring! I saw this just this evening, but I did not have time to fix it.

Reply:
I'm not sure I saw the part that remove the item from the inventory after usage. Did I miss it?

Reply:
It's in line 23 of Inventory.